packages/noir-ethereum-api/src/noir/oracles/accountOracles.ts

```diff
@@ -54,10 +60,13 @@ export function encodeAccount(ethProof: GetProofReturnType): ForeignCallOutput[]
 }
 
 function encodeProof(proof: string[]): string[] {
-  return proof
+  const encodedUnpaddedProof = proof
     .map((it) => encodeHex(it))
     .map((it) => padArray(it, PROOF_ONE_LEVEL_LENGTH, ZERO_PAD_VALUE))
     .reduce((accumulator, current) => accumulator.concat(current), []);
+  const encodedProof = padArray(encodedUnpaddedProof, PROOF_LENGTH, ZERO_PAD_VALUE);
+  // NOTE(Leo): We need to pad the proof to the length it has in the circuit.
```

Comment:
It wasn't obvious for me because I'm not used to writing noir, but I agree that it's obvious in the noir context as everything needs to have predefined length.

As for the NOTE(Leo) part. I've seen this system being used in some codebases but don't have a strong opinion about it:
* Leo: When someone reads this comment in 6m - they know who initially wrote it and who maybe have more context if they need it. Git blame may not be that useful as you may need to go a couple of levels back to see the original author and not someone who moved it in the refactor.
* NOTE:  You can prefix your comments with NOTE, HACK, TODO, OPTIMIZE to give a reader an initial context of what the comment if about. It's like a title/category. You can also use it later in greps. It can help you answer some questions like. "This thing runs slow. I remember - I've made suboptimal choice somewhere. Let me jump to that place and see if it's actually the issue" or "We have a code debt sprint. What are some small things in my part that could be fixed?".

In this context it makes **no sense** as the comment is obvious so I'll just remove it. Added because of muscle memory.

Reply:
:) I would propose to use comments rarely and embrace "now or never" attitude. 
We fix it in now it this PR (or following ticket/PR if to big of a step) or never.
Never:
Some things we can live with - let's agree on not fixing it now, and that perhaps means never 
Now:
Some things will be obvious - like big awaiting big features to be developed - we store those in tickets.
Some are small - let's do it right away.
Some are mid size - let's create a ticket and do it in the same sprint.
Tricky:
Some will be tricky - let's leave a comment for it, but let's try to write code that is self explanatory first.

Reply:
Agree. Removed and moved philosophy to the notion doc